src/ImageGallery.jsx

```diff
@@ -275,6 +278,10 @@ export default class ImageGallery extends React.Component {
       this.handleResize();
     }
     if (prevState.currentIndex !== currentIndex) {
+      if (onBeforeSlide) {
+        onBeforeSlide(currentIndex);
```

Comment:
Its easier to follow the logic if `onBeforeSlide` is placed in the `slideToIndex` function right before `setState` sets `nextIndex`

Reply:
Just be sure to make sure the index actually changed `nextIndex !== currentIndex`

Reply:
Great idea—pushing another commit in a second.